crates/wast/src/parser.rs

```diff
@@ -568,6 +578,38 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Same as the [`Parser::peek2`] method, except checks the next next token,
+    /// not the next token.
+    pub fn peek3<T: Peek>(self) -> bool {
+        let mut cursor = self.cursor();
+        if cursor.advance_token().is_some() && cursor.advance_token().is_some() {
+            T::peek(cursor)
+        } else {
+            false
+        }
+    }
+
+    /// Same as the [`Parser::peek3`] method, except checks the next next next
+    /// token, not the next next token.
+    pub fn peek4<T: Peek>(self) -> bool {
```

Comment:
Were these added opportunistically or was this needed for parsing somewhere internally? It wasn't my intention to really go infinitely long here with the peeking, and ideally we wouldn't add the desire to go too much further than the current spec needs.

Reply:
It was used in earlier version of this PR, as the text format has some tricky parsing ambiguities, but it's not needed anymore. I've now removed it.